quran/views.py

```diff
@@ -0,0 +1,36 @@
+from rest_framework import generics
+from rest_framework.response import Response
+from rest_framework.views import status
+
+from . import serializers
+from .models import Surah
+
+
+class SurahListView(generics.ListAPIView):
+    """
+    Provides a get method handler for Surah list.
+    """
+    queryset = Surah.objects.all()
+    serializer_class = serializers.SurahSerializer
+
+
+class SurahDetailsView(generics.RetrieveAPIView):
+    """
+    Provides a get method handler for Surah details.
+    """
+    queryset = Surah.objects.all()
+    serializer_class = serializers.SurahSerializer
+
+    def get(self, request, *args, **kwargs):
```

Comment:
You don't need to do it. Redefine just `get_queryset` method. It should be enough.
You can also use `get_object_or_404` method from `django.shortcuts`.

Reply:
I guess you don't need this method at all.
Once you've inherited from `generics.RetrieveAPIView` that's enough. It's done in backgoround.
just add to class attribute  ` lookup_field = 'surah_id'` (by default it's looking for `pk` or `id` so if you change
 r'quran/(?P<surah_id>\d+)/$',   surah_id to pk then no need to define lookup_field.

Just google about it (lookup field DRF etc) you wil find more info.